app/download/views.py

```diff
@@ -81,18 +85,33 @@ def check_document_exists(service_id, document_id):
         return jsonify(error='Invalid decryption key'), 400
 
     try:
-        document_exists = document_store.check_document_exists(service_id, document_id, key)
+        metadata = document_store.get_document_metadata(service_id, document_id, key)
     except DocumentStoreError as e:
         current_app.logger.warning(
-            'Failed to check if document exists: {}'.format(e),
+            'Failed to get document metadata: {}'.format(e),
             extra={
                 'service_id': service_id,
                 'document_id': document_id,
             }
         )
         return jsonify(error=str(e)), 400
 
-    response = make_response({'file_exists': str(document_exists)})
-    response.headers['X-Robots-Tag'] = 'noindex, nofollow'
+    if metadata:
+        document = {
+            'direct_file_url': get_direct_file_url(
+                service_id=service_id,
+                document_id=document_id,
+                key=key,
+                mimetype=metadata['mimetype'],
+            )
+        }
+    else:
+        document = None
 
+    response = make_response({
+        'file_exists': str(bool(metadata)),
```

Comment:
Yeah, it puzzled me why it was a string in the first place, since JSON supports booleans. It won't matter for much longer, though, since my other PR stops using this: https://github.com/alphagov/document-download-frontend/pull/86/commits/292787fbb49e624d5e08324a985120dcb8c06aa8

Reply:
(I've also added a note on the card so I remember to remove the flag later.)